internal/transform/scope-html.go

```diff
@@ -33,10 +35,32 @@ var NeverScopedElements map[string]bool = map[string]bool{
 func injectScopedClass(n *tycho.Node, opts TransformOptions) {
 	for i, attr := range n.Attr {
 		// If we find an existing class attribute, append the scoped class
-		if attr.Key == "class" {
-			attr.Val = attr.Val + " astro-" + opts.Scope
-			n.Attr[i] = attr
-			return
+		if attr.Key == "class" || (n.Component && attr.Key == "className") {
+			switch attr.Type {
+			case tycho.ShorthandAttribute:
+				if n.Component {
+					attr.Val = attr.Key + ` + " astro-` + opts.Scope + `"`
+					attr.Type = tycho.ExpressionAttribute
```

Comment:
Switch `<Component {className} />` to `<Component className={className + " astro-XXXXX"} />`